modules/core/src/main/java/org/apache/ignite/spi/ExponentialBackoffTimeoutStrategy.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.spi;
+
+import java.util.Date;
+
+/**
+ * Strategy which incorporates retriable network operation, handling of totalTimeout logic.
+ * It increases startTimeout based on exponential backoff algorithm.
+ *
+ * If failure detection is enabled it relies on totalTimeout
+ * otherwise implements exponential backoff totalTimeout logic based on startTimeout, maxTimeout and retryCnt.
+ */
+public class ExponentialBackoffTimeoutStrategy implements TimeoutStrategy {
+    /** Initial startTimeout, ms. */
+    private final long startTimeout;
+
+    /** Max startTimeout of the next try, ms. */
+    private final long maxTimeout;
+
+    /** Max reconnections count. */
+    private final int reconCnt;
+
+    /** Total startTimeout, ms. */
+    private final long totalTimeout;
+
+    /** Start of operation to check totalTimeout. */
+    private final long start;
+
+    /** Current startTimeout, ms. */
+    private long currTimeout;
+
+    /** Backoff coeffient to calculate next timeout.*/
+    private double backoffCoeff;
+
+    /** Compute expected backoffConnTimeout delay based on startTimeout, maxTimeout and reconCnt and backoff coeffient. */
+    public static long maxBackoffTimeout(
+            long startTimeout,
+            long maxTimeout,
+            long reconCnt,
+            double backoffCoeff
```

Comment:
Do you expect backoffCoeff will ever change ?

Reply:
Yes, it can be configurable via TcpCommunicationSpi.

Reply:
Made static